src/Persistent/PersistentMockRegistry.php

```diff
@@ -2,17 +2,31 @@
 
 namespace Hammock\Persistent;
 
+use namespace HH\Lib\{C, Vec};
 use type Hammock\Interfaces\IDeactivatable;
 
 class PersistentMockRegistry {
+	const REGISTRY_SOFT_LIMIT = 100;
+	const PRUNE_INTERVAL = 10;
 	protected static vec<IDeactivatable> $registry = vec[];
+	protected static int $pruneAttempts = 0;
 
 	public static function register(IDeactivatable $persistentMock): void {
-		// NOTE: When the registry grows beyond a certain threshold,
-		// we can prune the registry by removing persistent mocks
-		// that have been manually deactivated. This will require
-		// an `isDeactivated` interface on `IDeactivatable`.
 		self::$registry[] = $persistentMock;
+
+		// When the registry grows beyond REGISTRY_SOFT_LIMIT size,
+		// we prune the registry every PRUNE_INTERVAL calls to register
+		// by removing persistent mocks that have been manually deactivated.
+		//
+		// If the number of items in the registry drops below REGISTRY_SOFT_LIMIT,
+		// the number of attempts is reset to zero.
+		if (C\count(self::$registry) >= self::REGISTRY_SOFT_LIMIT &&
```

Comment:
I think you have made a coding error here.
The first pass you prune and `++` the pruneAttempts.
The next pass, `self::$pruneAttempts % self::PRUNE_INTERVAL === 0` is false, so you fall into the else case.
The else resets pruneAtetmpts to 0.
Wouldn't this create a cycle like this `0 -> 1 -> 0 -> 1 -> 0` for pruneAttempts?

Reply:
Oh geez, thanks for pointing that out @lexidor. I didn't quite think that through. I just pushed up a fix so that it attempts every PRUNE_INTERVAL times while the registry is over the soft limit. I picked 10 for now as a constant, but I'm open to talking about that